### PR TITLE
Mango test failures

### PR DESCRIPTION
**Pull request by tonysun83**
## Overview
Mango tests fail due to flaky tests with index deletion. I cannot recreate locally.
@davisp and @garrensmith
mentioned that it could be due to the latest ddoc_cache changes and suggested
changing r=3 to see if that would help. However, we recently changed tests to use
n=1 as default for all dbs created. I wonder if having w=3 and n=1 could be the reason.
We get a 202 back, but the index isn't deleted in time when we go to retrieve it. So changing
w=3  to w=1 to see if that will help. If not, I may add a wait loop inside delete.

## Testing recommendations
Let's see if it passes travis.

## Related Issues or Pull Requests
https://github.com/apache/couchdb/issues/918

## Checklist

- [X] Code is written and works correctly;
- [X] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: src/mango/test/mango.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def create_index(self, fields, idx_type="json", name=None, ddoc=None,
         body = json.dumps(body)
         r = self.sess.post(self.path("_index"), data=body)
         r.raise_for_status()
+
```
Comment (tonysun83) [inline src/mango/test/mango.py]: whitespace

File: src/mango/test/mango.py [hunk 2]
```diff
@@ -165,13 +166,13 @@ def list_indexes(self, limit="", skip=""):
...
         path = ["_index", ddocid, idx_type, name]
-        r = self.sess.delete(self.path(path), params={"w":"3"})
+        r = self.sess.delete(self.path(path), params={"w": "1"})
         r.raise_for_status()
 
     def bulk_delete(self, docs):
         body = {
             "docids" : docs,
-            "w": 3
+            "w": 1
```
Comment (tonysun83) [inline src/mango/test/mango.py]: eh not sure if I need to do this for bulk_delete since it's only index deletion is causing issues

**Comment by janl** [comment]
Sadly, still got this on the last run now:

```
FAIL: test_bulk_delete (01-index-crud-test.IndexCrudTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/apache/couchdb/src/mango/test/01-index-crud-test.py", line 189, in test_bulk_delete
    assert idx["type"] != "json"
AssertionError: 
-------------------- >> begin captured logging << --------------------
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "DELETE /mango_test_35e24dfb06ab46e4891c7e853ab7f08c HTTP/1.1" 200 12
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "GET /mango_test_35e24dfb06ab46e4891c7e853ab7f08c HTTP/1.1" 404 58
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "PUT /mango_test_35e24dfb06ab46e4891c7e853ab7f08c?q=1&n=1 HTTP/1.1" 201 12
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index HTTP/1.1" 200 93
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index HTTP/1.1" 200 93
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index HTTP/1.1" 200 93
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "GET /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index?; HTTP/1.1" 200 577
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "POST /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index/_bulk_delete HTTP/1.1" 200 290
urllib3.connectionpool: DEBUG: http://127.0.0.1:15984 "GET /mango_test_35e24dfb06ab46e4891c7e853ab7f08c/_index?; HTTP/1.1" 200 267
--------------------- >> end captured logging << ---------------------
----------------------------------------------------------------------
Ran 273 tests in 337.500s
FAILED (SKIP=131, failures=1)
make[1]: *** [mango-test] Error 1
make[1]: Leaving directory `/home/travis/build/apache/couchdb'
make: *** [check] Error 2
```

**Comment by tonysun83** [comment]
@janl : I added more delays after creating and deleting the index. Hopefully that should resolve the presumed ddoc_cache issue. The failure I saw was a per_user issue, not related to mango. Let me know if we want to give this commit a try and squash and merge.

**Comment by janl** [comment]
@tonysun83 good find! I refactored peruser tests on master recently, and it looks like on slow CI, we can get 500 timeouts from CouchDB and my I didn’t account for that. Fix in https://github.com/apache/couchdb/pull/930 — I’m restarting the travis build, to see if it passes.

**Pull request merged**